custom/enikshay/reports/views.py

```diff
@@ -0,0 +1,37 @@
+from django.http.response import JsonResponse
+from django.utils.decorators import method_decorator
+from django.views.generic.base import View
+
+from corehq.apps.domain.decorators import login_and_domain_required
+from corehq.apps.locations.permissions import location_safe
+from corehq.apps.userreports.reports.filters.choice_providers import ChoiceQueryContext, LocationChoiceProvider
+from custom.enikshay.reports.utils import Report
+
+
+class LocationsView(View):
+
+    @method_decorator(login_and_domain_required)
+    @method_decorator(location_safe)
```

Comment:
@esoergel do you know if this will work with your new method decorator? Or should we wait for your PR to be merged?